plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -0,0 +1,210 @@
+<?php
+declare(strict_types=1);
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+use Automattic\WooCommerce\Admin\Features\Features;
+use Automattic\WooCommerce\Blocks\Utils\StyleAttributesUtils;
+
+/**
+ * CatalogSorting class.
+ */
+class AddToCartWithOptions extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'add-to-cart-with-options';
+
+	/**
+	 * Initializes the AddToCartWithOptions block and hooks into the `wc_add_to_cart_message_html` filter
+	 * to prevent displaying the Cart Notice when the block is inside the Single Product block
+	 * and the Add to Cart button is clicked.
+	 *
+	 * It also hooks into the `woocommerce_add_to_cart_redirect` filter to prevent redirecting
+	 * to another page when the block is inside the Single Product block and the Add to Cart button
+	 * is clicked.
+	 *
+	 * @return void
+	 */
+	protected function initialize() {
+		parent::initialize();
+		add_filter( 'wc_add_to_cart_message_html', array( $this, 'add_to_cart_message_html_filter' ), 10, 2 );
+		add_filter( 'woocommerce_add_to_cart_redirect', array( $this, 'add_to_cart_redirect_filter' ), 10, 1 );
+	}
+
+	/**
+	 * Get the block's attributes.
+	 *
+	 * @param array $attributes Block attributes. Default empty array.
+	 * @return array  Block attributes merged with defaults.
+	 */
+	private function parse_attributes( $attributes ) {
+		// These should match what's set in JS `registerBlockType`.
+		$defaults = array(
+			'isDescendentOfSingleProductBlock' => false,
+		);
+
+		return wp_parse_args( $attributes, $defaults );
+	}
+
+	/**
+	 * Extra data passed through from server to client for block.
+	 *
+	 * @param array $attributes  Any attributes that currently are available from the block.
+	 *                           Note, this will be empty in the editor context when the block is
+	 *                           not in the post content on editor load.
+	 */
+	protected function enqueue_data( array $attributes = [] ) {
+		parent::enqueue_data( $attributes );
+		$this->asset_data_registry->add( 'isBlockifiedAddToCart', Features::is_enabled( 'blockified-add-to-cart' ) );
```

Comment:
no relevant to this PR, but I think the features should be exposed by another process.

Reply:
+1 to this! Should we create an issue for that? Are you thinking about something similar to what we do in [the frontend](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/Features/Features.php#L320)?

Reply:
That's a nice idea. or can we perform an async request?
Feel free to create an issue, but this is out of the scope of this block project